internal/clients/interface.go

```diff
@@ -57,3 +60,13 @@ type EC2 interface {
 	// RunInstances launches one or more instances
 	RunInstances(ctx context.Context, name *string, amount int32, instanceType types.InstanceType, AMI string, keyName string, userData []byte) ([]*string, *string, error)
 }
+
+var GetGCPClient func(ctx context.Context) (GCP, error)
+
+type GCP interface {
+	// Caller is responsible for closing the client using Close() call
+	Close()
```

Comment:
I suggest to move the comment from here to the variable `GetGCPClient` above. That is what you see in your editor/IDE when you actually use the function. Here you can just say something like "Close performs close on the gRPC client" or something like that.